### PR TITLE
Add token info to whoami

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Fixes https://github.com/pulumi/pulumi-cloud-requests/issues/236

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [ ] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/backend/backend.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ type Backend interface {
 	// LogoutAll logs you out of all the backend and removes any stored credentials.
 	LogoutAll() error
 	// Returns the identity of the current user and any organizations they are in for the backend.
-	CurrentUser() (string, []string, error)
+	CurrentUser() (string, []string, *workspace.TokenInformation, error)
```
Comment (dixler) [inline pkg/backend/backend.go:215]: Can we change the return type to `Account, error` since it would simplify producing and using the result?

Comment (Frassle) [reply pkg/backend/backend.go:215]: We _could_ but we need to return this from filestate as well which makes `Account` look a little odd given the only field we can fill in is "Username".

Comment (dixler) [reply pkg/backend/backend.go:215]: `CurrentUser()` already seems like a very httpstate backend specific method (filestate backend is returning the current OS user) and consolidating the return values would make the call-sites more consistent.

Comment (Frassle) [reply pkg/backend/backend.go:215]: This turned into a rabbit hole. Going to suggest sticking with it as in this PR to get his over the line, but to revisit when we've got some time to look into it again.

Comment (dixler) [reply pkg/backend/backend.go:215]: SGTM

**Review by dixler — APPROVED** [review]
LGTM

**Pull request merged**